src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,155 @@
+package duke.task;
+
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+
+import duke.command.CommandException;
+import duke.storage.LoadException;
+import duke.utils.Parser;
+
+/**
+ * Class to track and manage tasks.
+ *
+ * @author KohGuanZeh
+ */
+public class TaskList {
+    // Store list of all tasks.
+    private ArrayList<Task> taskList;
+
+    /**
+     * Creates a new instance of TaskList.
+     * Initial capacity is set to 100.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>(100);
+    }
+
+    /**
+     * Adds the specified Task and returns a message of the added Task.
+     *
+     * @param task Task to add.
+     * @return Message stating the added task.
+     */
+    public String addTask(Task task) {
+        this.taskList.add(task);
+        return "Added: " + task.getTaskInformation() + "\n" + this.getTotalTasks();
+    }
+
+    /**
+     * Removes the task at specified index and returns a message of the deleted Task.
+     *
+     * @param index Index of task as displayed.
+     * @return Message stating the deleted task.
+     */
+    public String deleteTask(int index) throws CommandException {
+        if (index <= 0 || index > this.taskList.size()) {
+            throw new CommandException("Error. Task of index " + index + " cannot be found.");
+        }
+        Task removedTask = this.taskList.remove(index - 1);
+        return "Removed: " + removedTask.getTaskInformation() + "\n" + this.getTotalTasks();
+    }
+
+    /**
+     * Returns the total number of tasks tracked.
+     *
+     * @return Message of total number of tasks tracked.
+     */
+    public String getTotalTasks() {
+        return "There are now " + this.taskList.size() + " tasks in the list.";
+    }
+
+    /**
+     * Sets the task at the specified index as done and returns a task completion message.
+     *
+     * @param index Index of task as displayed.
+     * @return Task completion message.
+     */
```

Comment:
Perhaps you can consider adding `@throws` in the Javadoc as there is an exception thrown in the header.